app/models/pet.rb

```diff
@@ -19,4 +19,13 @@ def self.pending
   def self.applied_for
     Pet.distinct.joins(:application_pets).order(:id)
   end
+
+  def adopter
+    approved_app = application_pets.find_by(approved: true)
+    if approved_app
+      approved_app.application.name
+    else
+      "nobody"
```

Comment:
added this edge case in case a pet has its status set to "Pending" without actually being applied for (this is the case for the another model test in `Pet`, shouldn't be possible in the development/production environment)